Customizable config file with ability to use local aws-proxy to access OpenSearch

johnnyon-amzn opened the pull request:
### Description
Set environment variables based on local config file and access OpenSearch with proxy credentials
### Issues Resolved
https://github.com/opensearch-project/dashboards-desktop/issues/11

johnnyon-amzn commented:
Squashing for readability. Electron app with customizable config and ability to connect to aws proxy

seraphjiang commented:
Great progress,
a few comments in general

1. move the code to src folder instead of put on project root folder
2. could we add a few screen shot about what look UI looks like so far

zhongnansu commented on renderer.js:
```diff
@@ -0,0 +1,48 @@
+    return [{id: "awsAccessKeyId", key: "AWS_ACCESS_KEY_ID"},
+            {id: "awsSecretAccessKey", key: "AWS_SECRET_ACCESS_KEY"},
+            {id: "endpoint", key: "ENDPOINT"},
+            {id: "osdPath", key: "osdPath"}
```
Do we need extra logic to validate that the path provided is actually an OSD path? What if user put some other random links or other aws resources links

johnnyon-amzn replied:
I will look into adding validation. I think the simplest would be to try to run the curl command and just output the stderr.

zhongnansu reviewed (commented):
Shouldn't we add license headers to all files?

johnnyon-amzn commented on README.md:
```diff
@@ -1,17 +1,32 @@
-* Edit your repository description on GitHub
+Dashboards Desktop offers desktop experiene to OpenSearch Dashboards Users. OpenSearch Dashboards is an open source search and analytics visualization application. 
+
+Dashboards Desktop provides end user *flexibility* and *extensibility* to install any Dashboards pluings they want. Dashboards Desktop could store data into local storage or cloud drive. User has full control of the data they have.
```
Update in commit 8fb16b4

johnnyon-amzn commented on README.md:
```diff
@@ -1,17 +1,32 @@
+
+Dashboards Desktop could be downloaded from internet. Not only deliver stable version to customer in monthly/quarterly release cadence. It will deliver experimental versions.
+
+Deskboards has build-in data source selector which allow user to connect to any OpenSearch endpoint they want with pre-configured credential or access key.
```
Update in commit 8fb16b4

johnnyon-amzn commented on config.json:
```diff
@@ -0,0 +1 @@
+{"AWS_ACCESS_KEY_ID":"","AWS_SECRET_ACCESS_KEY":"","ENDPOINT":"","osdPath":""}
```
Update in commit 8fb16b4

johnnyon-amzn commented:
> Great progress, a few comments in general
> 
> 1. move the code to src folder instead of put on project root folder
> 2. could we add a few screen shot about what look UI looks like so far

I migrated all code to the src folder, and changed the paths in commit 0924bed, 8fb16b4
I also added a screenshot for the initial page in commit 53c9349. I will add more screenshots for the next commits with opensearch dashboards connecting.

seraphjiang reviewed (approved):
Consider move css folder to src as well

Flyingliuhub commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+node_modules
+.DS_Store
+config.json
```
Is there any reason that we need to ignore config.json file? Thanks

johnnyon-amzn replied:
Config file has sensitive information like AWS_SECRET_KEY. I created a template file as a placeholder

Flyingliuhub commented on package.json:
Is it possible not have personal information "johnnyon-amzn"? I thought that should point to "opensearch-project/dashboards-desktop", the same as other url.

johnnyon-amzn replied:
I updated in commit a1eb3eb to reference main github page

zhongnansu reviewed (approved):
lgtm
